modules/ROOT/pages/upgrade-index.adoc

```diff
@@ -1,35 +1,282 @@
+
+. On the terminal open to the controller node that was used to start the upgrade, navigate to the directory with the installer bundle files, as shown in the following example:
++
+----
+cd /opt/anypoint/runtimefabric/installer
+----
++
+. Run the following command to resume the upgrade: 
++
+----
+sudo ./gravity upgrade --resume
+----
++
+. The upgrade procedure streams output to your terminal session. 
+
+If the previous error reoccurs, perform the troubleshooting steps described in the following section.
+ 
+== Troubleshooting Component-level (Appliance) Upgrade Errors
+
+A specific sequence of steps is performed during a cluster-level component (installer package) upgrade. If an error occurs, the upgrade pauses and displays an error. In most cases, the availability of running applications is not impacted when running multiple replicas of each application on a production Runtime Fabric configuration.
 
-## Applying upgrades
+Most errors encountered are due to insufficient disk performance on the `etcd` block device running on the controller nodes. 
 
-Upgrades to the cluster-level components and node configuration components are performed using the `rtfctl` command-line utility. Upgrades to the Runtime Fabric components are performed in Anypoint Runtime Manager.
+Perform the following steps to resolve common errors:
 
-== See Also
+. Use the `gravity plan` command to identify the phase in which the upgrade paused: 
++
+----
+sudo ./gravity plan
+----
++
+The following partial list provides upgrade phase examples:
++
+----
+* init 
+* checks
+* bootstrap
+  * node-1
+* masters
+  * node-1   
+    * drain
+    * system-upgrade
+    * taint
+    ...
+* runtime
+  * rbac-app
+  * site
+  * kubernetes
+* app
+  * telekube     
+----
++
+. Resume the upgrade using the `--debug` flag with the phase in which the error occurred. The following example resumes an upgrade by restarting the `masters/node-1/drain` phase: 
++
+----
+sudo ./gravity upgrade --phase=/masters/node-1/drain --force --debug
+----
+. Wait for the command to run. If the command does not terminate with an error, resume the upgrade by running the following command:
++
+----
+sudo ./gravity upgrade --resume
+----
+. If the upgrade again terminates with an error:
+.. Read the logs to identify which node requires repair.
+.. Submit a ticket to MuleSoft support if additional assistance is required.
+. Open another terminal to the Runtime Fabric node identified in the error logs.
+. Repair the upgrade plan for the identified node in the terminal:
++
+----
+sudo gravity plan --repair
+----
++
+. On the controller node running the upgrade, manually run the failed phase:
++
+----
+sudo ./gravity plan execute --phase=< insert phase > --force --debug
+----
 
-* xref:upgrade-cluster.adoc[Perform Runtime Fabric Cluster Upgrades]
+If an error is returned, wait a few minutes and repeat the previous steps.
```

Comment:
Does this also need some mention of contacting support if the repeat does not work?